src/plots/mod.rs

```diff
@@ -0,0 +1,26 @@
+pub mod histogram;
+
+use crate::util::Result;
+pub use histogram::Histogram;
+use serde::{Deserialize, Serialize};
+use std::fmt::Debug;
+
+pub trait Plot {
+    type PlotDataMetadataType: Debug + PartialEq + Serialize;
+
+    fn to_vega_embeddable(
+        &self,
+        allow_interactions: bool,
```

Comment:
maybe add a parameter whether the plot shall be produced as a vega string or as a picture (scatter)